sailthru/sailthru_content/services/catalog_api_service.py

```diff
@@ -63,25 +63,18 @@ def get_courses(self):
         logger.debug('Get Courses called')
         return self._get_resource_from_api(self.api_client.courses(), COURSES_PAGE_SIZE, marketable=1)
 
-    def get_searchable_course_keys(self):
+    def get_searchable_course_run_keys(self):
         logger.debug('Get Searchable Courses called')
         searchable_course_keys = []
-        searchable_courses = self._get_resource_from_search_api(self.api_client.search.all.facets())
-        for course in searchable_courses:
-            if course.get('key') and course.get('content_type') != 'program':
-                searchable_course_keys.append(course['key'])
+        searchable_course_runs = self._get_resource_from_search_api(
+            self.api_client.search.all.facets(), content_type='courserun'
```

Comment:
Yes, there is a reason, as you can see https://prod-edx-discovery.edx.org/api/v1/course_runs/?marketable=1 returns 1300 courses whereas on marketing site the number of courses return are 1291 which is same as per count as well as courses on the https://prod-edx-discovery.edx.org/api/v1/search/all/facets/. So we are not sure which 9 marketable courses are not searchable on the marketing site and we are not sure in the future as well. So it seems like https://prod-edx-discovery.edx.org/api/v1/search/all/facets/ is our only reliable endpoint for fetching searchable courses.